### PR TITLE
R4R: Updates for upstream PR (community pool spend proposal)

### PR DESCRIPTION
**Pull request by cwgoes**
Ref https://github.com/cosmos/cosmos-sdk/pull/4329

- [x] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote tests
- [x] Updated relevant documentation (`docs/`)
- [x] Added a relevant changelog entry: `clog add [section] [stanza] [message]`
- [x] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by cwgoes** [comment]
I think `test_cover` is broken for unrelated reasons, and `make check` doesn't seem to run the CLI tests.

**Comment by alexanderbez** [comment]
Fixed `master` @cwgoes. Not sure if you need to rebase (on the SDK) or not.

**Comment by alexanderbez** [comment]
Blocked on the approval/merge of https://github.com/cosmos/cosmos-sdk/pull/4329.

File: cmd/gaiacli/main.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/mint"
 	mintclient "github.com/cosmos/cosmos-sdk/x/mint/client"
 	mintrest "github.com/cosmos/cosmos-sdk/x/mint/client/rest"
+	distrcli "github.com/cosmos/cosmos-sdk/x/params/client/cli"
```
Comment (alessio) [inline cmd/gaiacli/main.go]: This is imported already in the following line

**Review by alessio — CHANGES_REQUESTED** [review]
Left a comment. Integration tests are failing too

File: cmd/gaiacli/main.go (hunk 1 above)
Comment (cwgoes) [reply cmd/gaiacli/main.go]: Thanks!

**Review by alexanderbez — APPROVED** [review]

**Review by alessio — APPROVED** [review]

**Pull request merged**